### PR TITLE
Fix block and staking ledger ingestion

### PR DESCRIPTION
**Pull request by trevorbernard**
Fixes: https://github.com/Granola-Team/mina-indexer/issues/548 https://github.com/Granola-Team/mina-indexer/issues/610
* https://github.com/Granola-Team/mina-indexer/issues/548
* https://github.com/Granola-Team/mina-indexer/issues/610

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -56,54 +59,37 @@ impl MinaIndexer {
...
+            // This modifies the state
+            if let Err(e) = run(block_watch_dir, ledger_watch_dir, state).await {
+                error!("Error in server run: {}", e);
                 std::process::exit(1);
             }
         });
 
         Ok(Self {
-            _ipc_join_handle,
+            //            domain_socket_handle,
```
Comment (Isaac-DeFrain) [inline src/server.rs:85]: same

File: src/server.rs [hunk 2]
```diff
@@ -56,54 +59,37 @@ impl MinaIndexer {
...
             _witness_join_handle,
         })
     }
 
     pub async fn await_loop(self) {
         let _ = self._witness_join_handle.await;
-        let _ = self._ipc_join_handle.await;
+        //let _ = self.domain_socket_handle.await;
```
Comment (Isaac-DeFrain) [inline src/server.rs:92]: same

File: src/unix_socket_server.rs [hunk 3]
```diff
@@ -5,86 +5,61 @@ use crate::{
...
-                            *self.summary.write().await = Some(*state.summary);
-                            *self.store.write().await = state.store;
-                        },
-                    }
-                }
+/// Start the Unix domain server
+/// TODO: Handle when bind fails
+pub async fn start(server: UnixSocketServer) -> JoinHandle<()> {
+    server::remove_domain_socket();
+    let listener = UnixListener::bind(SOCKET_NAME).expect("FOOBAR");
```
Comment (Isaac-DeFrain) [inline src/unix_socket_server.rs:42]: Probably meant to clean this up

File: test
Comment (Isaac-DeFrain) [inline test:1]: Do we require these sleep time changes?

**Review by Isaac-DeFrain — CHANGES_REQUESTED** [review]
LGTM other than small comments/questions

File: Cargo.toml [hunk 4]
```diff
@@ -16,11 +16,8 @@ loose_deserialization = []
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-anyhow = { version = "1.0.81" }
-async-trait = "0.1.64"
+anyhow = { version = "1.0.80" }
```
[the remark this answers is not in the document]
Comment (trevorbernard) [reply Cargo.toml:19]: I rebased and this snuck in. I'll change it

[the remark this answers is not in the document]
Comment (jhult) [reply Cargo.toml]: Fixed.

**Comment by jhult** [comment]
LGTM. Huge props to @trevorbernard and @Isaac-DeFrain for their work on this.

**Review by Isaac-DeFrain — APPROVED** [review]

**Pull request merged**